pkg/libp2p/peermgr/config.go

```diff
@@ -0,0 +1,36 @@
+package peermgr
+
+import (
+	"time"
+
+	"github.com/benbjohnson/clock"
+)
+
+type config struct {
+	runInterval      time.Duration
+	bootstrapTimeout time.Duration
+	clock            clock.Clock
+}
+
+type Option func(c *config) error
```

Comment:
Was aiming to avoid a scenario where an option is added that needs to return an error and we call panic instead as a quick hack (we've done this in other places). Think I should modify this function sig and assume we'll "do the right thing" in the even an error is needed in the future?